src/lib/HST/Feature/CaseCompletion.hs

```diff
@@ -74,26 +74,27 @@ completeCase _ e@(S.Lit _ _)              = return e
 completeBindRhs :: (Members '[Env a, Fresh, GetOpt, Report] r, S.EqAST a)
                 => S.Binds a
                 -> Sem r (S.Binds a)
-completeBindRhs (S.BDecls _ decls) = do
+completeBindRhs (S.BDecls s decls) = do
   decls' <- mapM (applyCCDecl True) decls
-  return $ S.BDecls S.NoSrcSpan decls'
+  return $ S.BDecls s decls'
 
 getEqFromAlt :: Member Report r => S.Alt a -> Sem r (Eqs a)
 getEqFromAlt (S.Alt _ pat rhs _) = do
   expr <- expFromUnguardedRhs rhs
   return ([pat], expr)
 
 completeLambda :: (Members '[Env a, Fresh, GetOpt, Report] r, S.EqAST a)
-               => [S.Pat a]
+               => S.SrcSpan a
+               -> [S.Pat a]
                -> S.Exp a
                -> Bool
                -> Sem r (S.Exp a)
-completeLambda ps e insideLet = do
+completeLambda s ps e insideLet = do
```

Comment:
Nice, adding an additional parameter like in this case would help removing some of the remaining `NoSrcSpan`s too.